javagen/src/main/java/com/azure/autorest/util/returntype/ReturnTypeDescriptionAssembler.java

```diff
@@ -0,0 +1,143 @@
+            if (isTypeResponse(returnType.getTypeArguments()[0])) { // Mono<Response<?>>
+                assembledDesc = String.format(
+                    "%s on successful completion of {@link Mono}",
+                    handlerRegistry.getHandler(returnType.getTypeArguments()[0]).handle(description, returnType.getTypeArguments()[0], baseType)
+                );
+            } else {
+                if (description == null) {
+                    if (PrimitiveType.Void == baseType) { // Mono<Void>
+                        assembledDesc = String.format("A {@link %s} that completes when a successful response is received", returnType.getName());
+                    } else { // Mono<OtherType>
+                        assembledDesc = String.format("the response body on successful completion of {@link %s}", returnType.getName());
+                    }
+                } else { // Mono<T>
+                    assembledDesc = String.format("%s on successful completion of {@link %s}", description, returnType.getName());
+                }
+            }
+            return assembledDesc;
+        }
+
+        private boolean isTypeResponse(IType type) {
+            return type instanceof GenericType &&
+                Response.class.isAssignableFrom(getGenericClass((GenericType) type));
+        }
+    }
+
+    private class ResponseDescriptionHandler implements ReturnTypeDescriptionHandler<GenericType> {
+
+        @Override
+        public boolean accept(IType returnType) {
+            return returnType instanceof GenericType && Response.class.isAssignableFrom(getGenericClass((GenericType) returnType));
+        }
+
+        /*
+        Response<Void> - the {@link Response}
+        Response<T> - "something" along with {@link Response}
+        Response<OtherType> - the response body along with {@link Response}
+         */
+        @Override
+        public String handle(String description, GenericType returnType, IType baseType) {
+            String assembledDesc;
+            if (description == null) {
+                if (PrimitiveType.Void == baseType) { // Response<Void>
+                    assembledDesc = String.format("the {@link %s}", returnType.getName());
+                } else { // Response<OtherType>
+                    assembledDesc = String.format("the response body along with {@link %s}", returnType.getName());
+                }
+            } else { // Response<T>
+                assembledDesc = String.format("%s along with {@link %s}", description, returnType.getName());
+            }
+            return assembledDesc;
+        }
+    }
+
+    private Class<?> getGenericClass(GenericType type) {
+        String className = String.format("%s.%s", type.getPackage(), type.getName());
+        try {
+            return Class.forName(className);
+        } catch (ClassNotFoundException e) {
+            logger.error(String.format("class %s not found!", className), e);
+            throw new RuntimeException(e);
```

Comment:
It occurred to me that this is codegen, it does not really had to load all the classes that to be generated in runtime.

So you should not throw exception here. If class not found in runtime, you know it is unknown, log a warning, and that's all.